File: feature-select-language/src/main/java/com/andyanika/translator/feature/select_lang/SelectLanguageViewHolder.java
package com.andyanika.translator.feature.select_lang;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import com.andyanika.translator.common.models.UiLanguageModel;

import io.reactivex.subjects.Subject;

public class SelectLanguageViewHolder extends RecyclerView.ViewHolder {
    private TextView title;
    private TextView description;
    private View currentIcon;

    SelectLanguageViewHolder(View itemView, Subject<Integer> subject) {
        super(itemView);
        title = itemView.findViewById(R.id.title);
        description = itemView.findViewById(R.id.description);
        currentIcon = itemView.findViewById(R.id.view_current);
        itemView.setOnClickListener(v -> {
            subject.onNext(getAdapterPosition());
            subject.onComplete();
        });
    }

    public void bind(UiLanguageModel model) {
        title.setText(model.code.toString());
        description.setText(model.description);
        currentIcon.setVisibility(model.isCurrent ? View.VISIBLE : View.GONE);
    }
}

File: feature-select-language/src/main/kotlin/com/andyanika/translator/feature/select/SelectLanguageViewHolder.kt
package com.andyanika.translator.feature.select

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.andyanika.translator.feature.select.databinding.ListviewRowSelectLanguageBinding
import core.models.ui.DisplayLanguageModel

internal class SelectLanguageViewHolder(
    itemView: View,
    private val action: (DisplayLanguageModel) -> Unit
) : RecyclerView.ViewHolder(itemView) {

    fun bind(model: DisplayLanguageModel) = with(ListviewRowSelectLanguageBinding.bind(itemView)) {
        title.text = model.code.toString()
        description.text = model.description
        selectedIcon.visibility = if (model.isSelected) View.VISIBLE else View.GONE
        itemView.setOnClickListener { action.invoke(model) }
    }
}
